tests/settings/test_ens_name_purchase.py

```diff
@@ -0,0 +1,79 @@
+import allure
+import pytest
+from allure_commons._allure import step
+
+import configs
+import constants
+import driver
+from configs.timeouts import UI_LOAD_TIMEOUT_SEC
+from constants.wallet import WalletTransactions
+from gui.components.onboarding.before_started_popup import BeforeStartedPopUp
+from gui.components.onboarding.beta_consent_popup import BetaConsentPopup
+from gui.components.splash_screen import SplashScreen
+from gui.components.wallet.authenticate_popup import AuthenticatePopup
+from gui.components.wallet.send_popup import SendPopup
+from gui.screens.onboarding import KeysView, AllowNotificationsView, WelcomeToStatusView, BiometricsView
+from gui.screens.settings_ens_usernames import ENSRegisteredView
+
+
+@pytest.fixture
+def keys_screen(main_window) -> KeysView:
+    with step('Open Generate new keys view'):
+        if configs.system.IS_MAC:
+            AllowNotificationsView().wait_until_appears().allow()
+        BeforeStartedPopUp().get_started()
+        wellcome_screen = WelcomeToStatusView().wait_until_appears()
+        return wellcome_screen.get_keys()
+
+
+@allure.testcase('https://ethstatus.testrail.net/index.php?/cases/view/704597',
+                 'Settings -> ENS usernames: buy ENS name on testnet')
+@pytest.mark.case(704597)
+@pytest.mark.parametrize('user_account', [constants.user.user_with_funds])
+@pytest.mark.parametrize('ens_name', [pytest.param(constants.user.ens_user_name)])
+def test_ens_name_purchase(keys_screen, main_window, user_account, ens_name):
+    with step('Open import seed phrase view and enter seed phrase'):
+        input_view = keys_screen.open_import_seed_phrase_view().open_seed_phrase_input_view()
+        input_view.input_seed_phrase(user_account.seed_phrase, True)
+        profile_view = input_view.import_seed_phrase()
+        profile_view.set_display_name(user_account.name)
+
+    with step('Finalize onboarding and open main screen'):
+        details_view = profile_view.next()
+        create_password_view = details_view.next()
+        confirm_password_view = create_password_view.create_password(user_account.password)
+        confirm_password_view.confirm_password(user_account.password)
+        if configs.system.IS_MAC:
+            BiometricsView().wait_until_appears().prefer_password()
+        SplashScreen().wait_until_appears().wait_until_hidden()
+        if not configs.system.TEST_MODE:
+            BetaConsentPopup().confirm()
```

Comment:
don't we have all that in `main_window.py`?

Reply:
We only have sign_up method, but it's not possible to use it here. But I can add one more method to main_window file then, because actually we need this part of sign up method quite often in our tests (which starts from password creation)

Reply:
i think we need the general change - divide this sign up method to 2 actions to have them being used separately. we can do it in separate task, dont worry

Reply:
Yes, agree with you. After we divide it, we will also need to replace these steps to the method from main_window.py in several tests then (for example import seed phrase test)